Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Payment.hs

```diff
@@ -86,6 +88,24 @@ createOrder (driverId, merchantId) driverFeeId = do
       createOrderCall = Payment.createOrder merchantId -- api call
   DPayment.createOrderService commonMerchantId commonPersonId orderId createOrderReq createOrderCall
 
+getOrder ::
+  ( CacheFlow m r,
+    EsqDBReplicaFlow m r,
+    EsqDBFlow m r,
+    EncFlow m r,
+    CoreMetrics m
+  ) =>
+  (Id DP.Person, Id DM.Merchant) ->
+  Id DOrder.PaymentOrder ->
+  m DOrder.PaymentOrderAPIEntity
+getOrder (personId, _) orderId = do
```

Comment:
Handler can be moved in payment Lib. But anyway this can be refactored later

Reply:
Hmm, that's what I did earlier but `personId` will be coming from diff domains so we can make it generic later

Reply:
Then this handler should call common handler and use `cast`. Anyway, let it be for now